guides/guide-for-adding-windows-node.md

```diff
@@ -1,229 +1,402 @@
+```bash
+kubectl taint nodes --all node-role.kubernetes.io/control-plane-
+```
 
-  
+It should return the following.
+
+```
+node/<your-hostname> untainted
+```
+3. Add calico and kube-proxy daemonsets from sig-windows-tools/hostprocess/calico
+
+Now you can add Windows-compatible versions of calico and kube-proxy. In order to ensure that you get a compatible version of kube-proxy, you'll need to substitute the tag of the image. The following example shows usage for Kubernetes v1.24.3, but you should adjust the version for your own deployment.
 
 ```bash
+git clone https://github.com/kubernetes-sigs/sig-windows-tools
+# Make sure to modify the version of kube-proxy image to match your kubernetes version
+# before applying it
+kubectl apply -f sig-windows-tools/hostprocess/calico/kube-proxy/kube-proxy.yml
+kubectl apply -f sig-windows-tools/hostprocess/calico/calico.yml
+```
 
-curl -L https://github.com/kubernetes-sigs/sig-windows-tools/releases/latest/download/kube-proxy.yml | sed 's/VERSION/v1.24.3/g' | kubectl apply -f -
+>  **Note** To find your version of kubernetes run the following command:
+> `kubeadm version`
 
-kubectl apply -f https://github.com/kubernetes-sigs/sig-windows-tools/releases/latest/download/flannel-overlay.yml
+5. Apply kube-calico-rbac.yml from sig-windows-tools/kubeadm/flannel
+Next you will need to apply the configuration that allows calico to spawn pods and keep them running:
 
+```bash
+kubectl apply -f sig-windows-tools/hostprocess/calico/kube-calico-rbac.yml
 ```
 
->  **Note** If you are using another version of kubernetes on your windows node, change v1.24.3 with your own version .
 
+### Joining a Windows worker node
+
+>  **Note:** All code snippets in Windows sections are to be run in a PowerShell environment with elevated permissions (Administrator) on the Windows worker node.
+
+1. Install ContainerD, wins, kubelet, and kubeadm.
+
+```PowerShell
+git clone https://github.com/kubernetes-sigs/sig-windows-tools
+cd .\sig-windows-tools\kubeadm\scripts\
+
+# Install ContainerD
+.\Install-Containerd.ps1
+
+# Install wins, kubelet and kubeadm
+.\PrepareNode.ps1 -KubernetesVersion v1.24.3 -ContainerRuntime containerD
+```
+
+>  **Note** If you want to install another version of kubernetes, modify v1.24.3 with the version you want to install
+
+
+2. Run `kubeadm` to join the node
+
+Use the command that was given to you when you ran `kubeadm init` on a control plane host. If you no longer have this command, or the token has expired, you can run `kubeadm token create --print-join-command` (on a control plane host) to generate a new token and join command.
+
+>  **Note:** Do not forget to add `--cri-socket "npipe:////./pipe/containerd-containerd" --v=5` at the end of the join command, if you use ContainerD
+
+3. Install kubectl for windows (optional)
+
+For more information about it : https://kubernetes.io/docs/tasks/tools/install-kubectl-windows/
+
+
+#### Verifying your installation
```

Comment:
As the non-CNI-related steps on Linux and Windows as well as the verification steps are always the same, may think about consolidating them in a single paragraph instead of duplicating it multiple times